www/app/hardware/Hardware.html

```diff
@@ -1485,6 +1485,10 @@ <h2><span data-i18n="Hardware">Hardware</span>:</h2>
 					</div>
 				</td>
 			</tr>
+			<tr>
+				<td align="right" style="width:110px"><span data-i18n="Migration Tool"></span>:</label></td>
```

Comment:
I guess the migration is a onetime only thing to do?
If yes, this should not be here.
Maybe you could use a Preference version value (NetatmoHWVersion) and if this does not exists, or is a specific value, call the migration at startup of the Netatmo hardware?

Reply:
This way also no changes in the web part needs to be made

Reply:
Not really: It needs to run several times, depending on the moment that the "Early Adaptor" started using the new software (there were several changes over time in the device ID (incl. case changes), unit numbers, types, and subtypes during the development process). I planned to do this in one run, but realized that the "New" device (that is triggering the migration for this device) was changing during the process, and also, I needed to work backwards in time. So I needed to split up. Users that will step in now, probably will need only one run, but users that stepped in after the first update, probably need to run it 3 or even 4 times (as in my case).

I also think that I should give the user an option to use the tool or not (the default setting of the flag is OFF).

There is no changed hardware or firmware version to test on; All changes are caused by the new interface within Domoticz (in hardware/Netatmo.cpp).

The only way to test if a migration is useful, is the very complex SQL statement that matches old and new devices, based on their Device ID (or a part of it). That is why I wanted to introduce the flag, so it can be switched off after the conversion to avoid unnecessary database activity.

I also think that the whole tool, including the flag on the website, can be removed after one or two master releases.

Reply:
Yes, that's why I mentioned the idea of using a version number in the preferences so you can migrate on startup of the hardware

Reply:
But, in that case, I need to add an option to the preferences' webpage to enable the migration. While in BETA, I don't want to run this tool without permission and knowledge of the user.

Reply:
No, there are functions like

`CSQLHelper::GetPreferencesVar`

```
m_netamo_hw_version = 1;
if (!m_sql.GetPreferencesVar("NetamoHWVersion", m_netamo_hw_version)) m_netamo_hw_version = 0;

```

Reply:
That still leave me with two questions: How should the user enable the migration tool, and what if the user has more versions of the device running (e.g. one for production and one for testing like me, or one for the weather station and one for the alarm system)?

Can we please agree that I use this flag on the hardware page for beta testing, and that I replace it with your suggestion (or something like that) before the next master release?

Reply:
The user should 'not' be bothered with the migration tool. You as developer knows this.
Version 2 (this new version), we need to convert some sensors.
Later some more changes, version 3 (#define at the top of the class), you need to convert more

Reply:
OK, I rest my case: see the last commit (removed migration flag)